src/features/FreeRooms/FreeRooms.ts

```diff
@@ -198,18 +191,19 @@ async function getAvailableRooms(date) {
     return availableRooms;
 }
 
-chrome.runtime.onInstalled.addListener(details => {
-    getAvailableRooms(new Date()).then(rooms => {
-        chrome.storage.local.set({availableRooms: rooms});
-        chrome.alarms.create('rooms', {
-            periodInMinutes: 15,
-        });
+chrome.runtime.onInstalled.addListener(async details => {
+    let today = new Date();
+    let semester = getSemesterFromDate(today);
+    let html = parse(await fetchRoomSchedHTML(today.getFullYear(), semester));
+    chrome.storage.local.set({availableRooms: getAvailableRooms(today, html)});
+    chrome.alarms.create('rooms', {
+        periodInMinutes: 15,
     });
 
     chrome.alarms.onAlarm.addListener(alarm => {
         if (alarm.name === 'rooms') {
-            getAvailableRooms(new Date()).then(rooms => {
-                chrome.storage.local.set({availableRooms: rooms});
+            chrome.storage.local.set({
+                availableRooms: getAvailableRooms(today, html),
```

Comment:
`today` is instantiated on install. Should I instantiate a new `Date` for when the alarm event fires?

Reply:
I think the best way of handling this would be to move everything that is required to update the available rooms into the function itself. Specifically, these lines should be moved:
https://github.com/StevensSEC/stevens-web-extension/blob/a1789cea1577aff944f5444e011eaf57b974b61c/src/features/FreeRooms/FreeRooms.ts#L195-L198